--- a/net/minecraft/data/recipes/ShapedRecipeBuilder.java
+++ b/net/minecraft/data/recipes/ShapedRecipeBuilder.java
@@ -17,19 +_,24 @@
 import net.minecraft.world.level.ItemLike;
 import org.jspecify.annotations.Nullable;
 
-import net.neoforged.neoforge.common.extensions.IShapedRecipeBuilderExtension;
-
-public class ShapedRecipeBuilder implements RecipeBuilder, IShapedRecipeBuilderExtension {
+public class ShapedRecipeBuilder implements RecipeBuilder {
     private final HolderGetter<Item> items;
     private final RecipeCategory category;
-    private final ItemStackTemplate result;
+    private final net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result;
     private final List<String> rows = Lists.newArrayList();
     private final Map<Character, Ingredient> key = Maps.newLinkedHashMap();
     private final RecipeUnlockAdvancementBuilder advancementBuilder = new RecipeUnlockAdvancementBuilder();
     private @Nullable String group;
     private boolean showNotification = true;
+    private @Nullable ResourceKey<Recipe<?>> defaultId = null; // Neo: used in the non-outgredient constructor
 
     private ShapedRecipeBuilder(HolderGetter<Item> items, RecipeCategory category, ItemStackTemplate result) {
+        this(items, category, net.neoforged.neoforge.common.crafting.outgredient.Outgredient.of(result));
+        this.defaultId = RecipeBuilder.getDefaultRecipeId(result);
+    }
+
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public ShapedRecipeBuilder(HolderGetter<Item> items, RecipeCategory category, net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result) {
         this.items = items;
         this.category = category;
         this.result = result;
@@ -47,6 +_,10 @@
         return new ShapedRecipeBuilder(items, category, item, count);
     }
 
+    public static ShapedRecipeBuilder shaped(HolderGetter<Item> items, RecipeCategory category, ItemStackTemplate result) {
+        return new ShapedRecipeBuilder(items, category, result);
+    }
+
     public ShapedRecipeBuilder define(Character symbol, TagKey<Item> tag) {
         return this.define(symbol, Ingredient.of(this.items.getOrThrow(tag)));
     }
@@ -94,7 +_,7 @@
 
     @Override
     public ResourceKey<Recipe<?>> defaultId() {
-        return RecipeBuilder.getDefaultRecipeId(this.result);
+        return defaultId; // Neo: using `RecipeBuilder.getDefaultRecipeId(this.result);` is no longer feasible with outgredients in the mix
     }
 
     @Override
