--- a/net/minecraft/data/recipes/SimpleCookingRecipeBuilder.java
+++ b/net/minecraft/data/recipes/SimpleCookingRecipeBuilder.java
@@ -20,13 +_,14 @@
 public class SimpleCookingRecipeBuilder implements RecipeBuilder, ISimpleCookingRecipeBuilderExtension {
     private final RecipeCategory craftingCategory;
     private final CookingBookCategory cookingCategory;
-    private final ItemStackTemplate result;
+    private final net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result;
     private final Ingredient ingredient;
     private final float experience;
     private final int cookingTime;
     private final RecipeUnlockAdvancementBuilder advancementBuilder = new RecipeUnlockAdvancementBuilder();
     private @Nullable String group;
     private final AbstractCookingRecipe.Factory<?> factory;
+    private @Nullable ResourceKey<Recipe<?>> defaultId = null; // Neo: used in the non-outgredient constructor
 
     private SimpleCookingRecipeBuilder(
         RecipeCategory craftingCategory,
@@ -37,6 +_,19 @@
         int cookingTime,
         AbstractCookingRecipe.Factory<?> factory
     ) {
+        this(craftingCategory, cookingCategory, net.neoforged.neoforge.common.crafting.outgredient.Outgredient.of(result), ingredient, experience, cookingTime, factory);
+        this.defaultId = RecipeBuilder.getDefaultRecipeId(result);
+    }
+
+    private SimpleCookingRecipeBuilder(
+        RecipeCategory craftingCategory,
+        CookingBookCategory cookingCategory,
+        net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result,
+        Ingredient ingredient,
+        float experience,
+        int cookingTime,
+        AbstractCookingRecipe.Factory<?> factory
+    ) {
         this.craftingCategory = craftingCategory;
         this.cookingCategory = cookingCategory;
         this.result = result;
@@ -94,6 +_,45 @@
         return new SimpleCookingRecipeBuilder(craftingCategory, CookingBookCategory.FOOD, result, ingredient, experience, cookingTime, SmokingRecipe::new);
     }
 
+    public static <T extends AbstractCookingRecipe> SimpleCookingRecipeBuilder generic(
+            Ingredient ingredient,
+            RecipeCategory craftingCategory,
+            CookingBookCategory cookingCategory,
+            ItemStackTemplate result,
+            float experience,
+            int cookingTime,
+            AbstractCookingRecipe.Factory<T> factory
+    ) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, cookingCategory, result, ingredient, experience, cookingTime, factory);
+    }
+    public static SimpleCookingRecipeBuilder campfireCooking(Ingredient ingredient, RecipeCategory craftingCategory, ItemStackTemplate result, float experience, int cookingTime) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, CookingBookCategory.FOOD, result, ingredient, experience, cookingTime, CampfireCookingRecipe::new);
+    }
+
+    public static SimpleCookingRecipeBuilder blasting(Ingredient ingredient, RecipeCategory craftingCategory, CookingBookCategory cookingCategory, ItemStackTemplate result, float experience, int cookingTime) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, cookingCategory, result, ingredient, experience, cookingTime, BlastingRecipe::new);
+    }
+
+    public static SimpleCookingRecipeBuilder smelting(Ingredient ingredient, RecipeCategory craftingCategory, CookingBookCategory cookingCategory, ItemStackTemplate result, float experience, int cookingTime) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, cookingCategory, result, ingredient, experience, cookingTime, SmeltingRecipe::new);
+    }
+
+    public static SimpleCookingRecipeBuilder smoking(Ingredient ingredient, RecipeCategory craftingCategory, ItemStackTemplate result, float experience, int cookingTime) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, CookingBookCategory.FOOD, result, ingredient, experience, cookingTime, SmokingRecipe::new);
+    }
+
+    public static <T extends AbstractCookingRecipe> SimpleCookingRecipeBuilder generic(
+        Ingredient ingredient,
+        RecipeCategory craftingCategory,
+        CookingBookCategory cookingCategory,
+        net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result,
+        float experience,
+        int cookingTime,
+        AbstractCookingRecipe.Factory<T> factory
+    ) {
+        return new SimpleCookingRecipeBuilder(craftingCategory, cookingCategory, result, ingredient, experience, cookingTime, factory);
+    }
+
     public SimpleCookingRecipeBuilder unlockedBy(String name, Criterion<?> criterion) {
         this.advancementBuilder.unlockedBy(name, criterion);
         return this;
@@ -106,7 +_,7 @@
 
     @Override
     public ResourceKey<Recipe<?>> defaultId() {
-        return RecipeBuilder.getDefaultRecipeId(this.result);
+        return defaultId; // Neo: using `RecipeBuilder.getDefaultRecipeId(this.result);` is no longer feasible with outgredients in the mix
     }
 
     @Override
