--- a/net/minecraft/world/item/crafting/BrewingRecipe.java
+++ b/net/minecraft/world/item/crafting/BrewingRecipe.java
@@ -13,7 +_,7 @@
         i -> i.group(
                 PotionIngredient.MAP_CODEC.fieldOf("input").forGetter(o -> o.input),
                 PotionIngredient.MAP_CODEC.fieldOf("reagent").forGetter(o -> o.reagent),
-                ItemStackTemplate.CODEC.fieldOf("output").forGetter(o -> o.output)
+                net.neoforged.neoforge.common.crafting.outgredient.OutgredientCodecs.ITEM_OUTGREDIENT_CODEC.fieldOf("output").forGetter(o -> o.output)
             )
             .apply(i, BrewingRecipe::new)
     );
@@ -22,16 +_,20 @@
         o -> o.input,
         PotionIngredient.STREAM_CODEC,
         o -> o.reagent,
-        ItemStackTemplate.STREAM_CODEC,
+        net.neoforged.neoforge.common.crafting.outgredient.OutgredientCodecs.ITEM_OUTGREDIENT_STREAM_CODEC,
         o -> o.output,
         BrewingRecipe::new
     );
     public static final RecipeSerializer<BrewingRecipe> SERIALIZER = new RecipeSerializer<>(MAP_CODEC, STREAM_CODEC);
     private final PotionIngredient input;
     private final PotionIngredient reagent;
-    private final ItemStackTemplate output;
+    private final net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> output;
 
     public BrewingRecipe(PotionIngredient input, PotionIngredient reagent, ItemStackTemplate output) {
+        this(input, reagent, net.neoforged.neoforge.common.crafting.outgredient.Outgredient.of(output));
+    }
+
+    public BrewingRecipe(PotionIngredient input, PotionIngredient reagent, net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> output) {
         this.input = input;
         this.reagent = reagent;
         this.output = output;
@@ -46,7 +_,7 @@
     }
 
     public ItemStackTemplate getOutput() {
-        return this.output;
+        return this.output.resolve();
     }
 
     public boolean matches(BrewingInput brewingInput, Level level) {
@@ -58,7 +_,7 @@
     }
 
     public ItemStack assemble(BrewingInput input) {
-        return this.output.create();
+        return this.output.resolve().create();
     }
 
     @Override
