--- a/net/minecraft/world/item/crafting/CampfireCookingRecipe.java
+++ b/net/minecraft/world/item/crafting/CampfireCookingRecipe.java
@@ -23,6 +_,17 @@
         super(commonInfo, bookInfo, ingredient, result, experience, cookingTime);
     }
 
+    public CampfireCookingRecipe(
+        Recipe.CommonInfo commonInfo,
+        AbstractCookingRecipe.CookingBookInfo bookInfo,
+        Ingredient ingredient,
+        net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result,
+        float experience,
+        int cookingTime
+    ) {
+        super(commonInfo, bookInfo, ingredient, result, experience, cookingTime);
+    }
+
     @Override
     protected Item furnaceIcon() {
         return Items.CAMPFIRE;
