--- a/net/minecraft/world/item/crafting/ShapelessRecipe.java
+++ b/net/minecraft/world/item/crafting/ShapelessRecipe.java
@@ -19,8 +_,8 @@
         i -> i.group(
                 Recipe.CommonInfo.MAP_CODEC.forGetter(o -> o.commonInfo),
                 CraftingRecipe.CraftingBookInfo.MAP_CODEC.forGetter(o -> o.bookInfo),
-                ItemStackTemplate.CODEC.fieldOf("result").forGetter(o -> o.result),
-                Ingredient.CODEC.listOf(1, 9).fieldOf("ingredients").forGetter(o -> o.ingredients)
+            net.neoforged.neoforge.common.crafting.outgredient.OutgredientCodecs.ITEM_OUTGREDIENT_CODEC.fieldOf("result").forGetter(o -> o.result),
+                com.mojang.serialization.Codec.lazyInitialized(() -> Ingredient.CODEC.listOf(1, ShapedRecipePattern.maxHeight * ShapedRecipePattern.maxWidth)).fieldOf("ingredients").forGetter(o -> o.ingredients)
             )
             .apply(i, ShapelessRecipe::new)
     );
@@ -29,20 +_,26 @@
         o -> o.commonInfo,
         CraftingRecipe.CraftingBookInfo.STREAM_CODEC,
         o -> o.bookInfo,
-        ItemStackTemplate.STREAM_CODEC,
+        net.neoforged.neoforge.common.crafting.outgredient.OutgredientCodecs.ITEM_OUTGREDIENT_STREAM_CODEC,
         o -> o.result,
         Ingredient.CONTENTS_STREAM_CODEC.apply(ByteBufCodecs.list()),
         o -> o.ingredients,
         ShapelessRecipe::new
     );
     public static final RecipeSerializer<ShapelessRecipe> SERIALIZER = new RecipeSerializer<>(MAP_CODEC, STREAM_CODEC);
-    private final ItemStackTemplate result;
+    private final net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result;
     private final List<Ingredient> ingredients;
+    private final boolean isSimple;
 
     public ShapelessRecipe(Recipe.CommonInfo commonInfo, CraftingRecipe.CraftingBookInfo bookInfo, ItemStackTemplate result, List<Ingredient> ingredients) {
+        this(commonInfo, bookInfo, net.neoforged.neoforge.common.crafting.outgredient.Outgredient.of(result), ingredients);
+    }
+
+    public ShapelessRecipe(Recipe.CommonInfo commonInfo, CraftingRecipe.CraftingBookInfo bookInfo, net.neoforged.neoforge.common.crafting.outgredient.Outgredient<ItemStackTemplate> result, List<Ingredient> ingredients) {
         super(commonInfo, bookInfo);
         this.result = result;
         this.ingredients = ingredients;
+        this.isSimple = ingredients.stream().allMatch(Ingredient::isSimple);
     }
 
     @Override
@@ -58,6 +_,12 @@
     public boolean matches(CraftingInput input, Level level) {
         if (input.ingredientCount() != this.ingredients.size()) {
             return false;
+        } else if (!isSimple) {
+            var nonEmptyItems = new java.util.ArrayList<ItemStack>(input.ingredientCount());
+            for (var item : input.items())
+                if (!item.isEmpty())
+                    nonEmptyItems.add(item);
+            return net.neoforged.neoforge.common.util.RecipeMatcher.findMatches(nonEmptyItems, this.ingredients) != null;
         } else {
             return input.size() == 1 && this.ingredients.size() == 1
                 ? this.ingredients.getFirst().test(input.getItem(0))
@@ -65,8 +_,13 @@
         }
     }
 
+    /// {@return the result of this shapeless recipe or null if it is not static and needs to be obtained by assembling it}
+    public @org.jspecify.annotations.Nullable ItemStackTemplate result() {
+        return result.resolve();
+    }
+
     public ItemStack assemble(CraftingInput input) {
-        return this.result.create();
+        return this.result.resolve().create();
     }
 
     @Override
@@ -74,7 +_,7 @@
         return List.of(
             new ShapelessCraftingRecipeDisplay(
                 this.ingredients.stream().map(Ingredient::display).toList(),
-                new SlotDisplay.ItemStackSlotDisplay(this.result),
+                this.result.display(),
                 new SlotDisplay.ItemSlotDisplay(Items.CRAFTING_TABLE)
             )
         );
